tests/Microsoft.Bot.Builder.Tests/MentionTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Adapters;
+using Microsoft.Bot.Schema;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Tests
+{
+    [TestClass]
+    [TestCategory("Mention")]
+    public class MentionTests
+    {
+        [TestMethod]
+        public async Task Mention_SkypeAsync()
+        {
+            TestAdapter adapter = new TestAdapter();
+            adapter.Use(new SkypeMiddleware());
+            adapter.Conversation = new ConversationReference
+            {
+                ChannelId = "skype",
+                ServiceUrl = "https://test.com",
+                User = new ChannelAccount("user1", "User1"),
+                Bot = new ChannelAccount("bot", "Bot"),
+                Conversation = new ConversationAccount(false, "convo1", "Conversation1"),
+            };
+
+            // A Skype mention contains the user mention enclosed in <at> tags.  But the Activity.Text (as below)
+            // does not.  The SkpeMiddleware will modify the Mention.Text to just contain the user so that
+            // RemoveMentionText works.
+            var mentionJson = "{\"mentioned\": {\"id\": \"recipientid\"},\"text\": \"<at id=\\\"28: 841caffa-9e92-425d-8d84-b503b3ded285\\\">botname</at>\"}";
+            var mention = JsonConvert.DeserializeObject<Entity>(mentionJson);
+            mention.Type = "mention";
+
+            var activity = MessageFactory.Text("botname sometext");
+            activity.Entities.Add(mention);
+
+            await adapter.ProcessActivityAsync(activity, async (turnContext, cancellationToken) =>
+            {
+                turnContext.Activity.RemoveMentionText("recipientid");
```

Comment:
Maybe it's the lack of coffee today, but I don't understand the relationship between this call and the actual middleware. 

I'm *guessing* the middleware puts the Skype mention in a standard format so the RemoveMentionText call actually works. Can you confirm?

Reply:
Correct.  The middleware just gets the mention Entity into a conforming format so that RemoveMentionText will work against it the same way it does the others.